nixos/modules/services/torrent/deluge.nix

```diff
@@ -20,9 +60,93 @@ in {
             Number of files to allow deluged to open.
           '';
         };
+
+        coreConf = mkOption {
+          type = types.attrs;
+          example = {
+            download_location = "/srv/torrents/";
+            max_upload_speed = "1000.0";
+            share_ratio_limit = "2.0";
+            allow_remote = true;
+          };
+          description = ''
+            Deluge core configuration for the core.conf file. Only has
+            an effect when <option>services.deluge.declarative</option>
+            is set to <literal>true</literal>.
+            String values must be quoted, integer and boolean values must not.
+          '';
+        };
+
+      declarative = mkOption {
+        type = types.bool;
+        default = false;
+        description = ''
+          Whether to use a declarative deluge configuration.
+          Only if set to <literal>true</literal>, the options
+          <option>services.deluge.coreConf</option>,
+          <option>services.deluge.listenPortRange</option> and
+          <option>services.deluge.auth</option> will be
+          applied.
+        '';
       };
 
-      deluge.web.enable = mkEnableOption "Deluge Web daemon";
+        port = mkOption {
+        type = types.ints.u16;
+          default = 58846;
+          description = ''
+            Deluge daemon port.
+          '';
+        };
+
+        listenPortRange = mkOption {
+          type = types.listOf types.attrs;
+          default = [ { from = 56881; to = 56889; } ];
```

Comment:
This also doesn't need an option, the user can set it through `coreConf.listen_ports = [ 56881 56889 ]`. And btw, it doesn't look like those are the defaults, my daemon uses `6881-6891` as the default

Reply:
It has been removed.

Reply:
And the default port changed.